### PR TITLE
E2E test for GNMI native write

### PR DESCRIPTION
**Pull request by ganglyu**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?
And e2e test for GNMI native write.

#### How did you do it?
Create certificates for GNMI with below tests:
GNMI capabilities.
Incremental config update for ConfigDB.
Full config update for ConfigDB.
Update and delete for ApplDB.

#### How did you verify/test it?
Run sonic-mgmt test.

### CODE REVIEW

File: tests/gnmi/conftest.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+from helper import GNMI_CONTAINER_NAME, apply_cert_config, create_ext_conf
+from tests.generic_config_updater.gu_utils import create_checkpoint, rollback
+
+SETUP_ENV_CP = "test_setup_checkpoint"
+
+
+@pytest.fixture(scope="function", autouse=True)
+def skip_non_x86_platform(duthosts, rand_one_dut_hostname):
+    """
+    Skip the current test if DUT is not x86_64 platform.
```
Comment (qiluo-msft) [inline tests/gnmi/conftest.py:15]: >x86_64 [](http://example.com/codeflow?start=40&length=6)

Is it possible to test other platforms?

File: tests/gnmi/conftest.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    platform = duthost.facts["platform"]
+    if 'x86_64' not in platform:
+        pytest.skip("Test not supported for current platform. Skipping the test")
+
+
+@pytest.fixture(scope="module", autouse=True)
+def download_gnmi_client(duthosts, rand_one_dut_hostname, localhost):
```
Comment (qiluo-msft) [inline tests/gnmi/conftest.py:24]: >download_gnmi_client [](http://example.com/codeflow?start=4&length=20)

telemetry is tested with ptf "gnmi_cli_py/py_gnmicli.py". Is it better than download_gnmi_client?

File: tests/gnmi/helper.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
+import time
+import re
+
+GNMI_CONTAINER_NAME = 'telemetry'
+GNMI_PROGRAM_NAME = 'telemetry'
+GNMI_CONFIG_KEY = 'TELEMETRY|gnmi'
+GNMI_SERVER_START_WAIT_TIME = 15
```
Comment (qiluo-msft) [inline tests/gnmi/helper.py:7]: >15 [](http://example.com/codeflow?start=30&length=2)

Add comment about time unit? In seconds?

File: tests/gnmi/test_gnmi_appldb.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import logging
+import pytest
+
+from helper import gnmi_set, gnmi_get
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('t0'),
```
Comment (qiluo-msft) [inline tests/gnmi/test_gnmi_appldb.py:9]: >t0 [](http://example.com/codeflow?start=26&length=2)

Why only T0?

File: tests/gnmi/test_gnmi_configdb.py [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+from helper import gnmi_set, gnmi_get, gnoi_reboot
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.utilities import wait_until
+from tests.common.platform.processes_utils import wait_critical_processes
+from tests.common.platform.interface_utils import check_interface_status_of_up_ports
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('t0'),
```
Comment (qiluo-msft) [inline tests/gnmi/test_gnmi_configdb.py:15]: >t0 [](http://example.com/codeflow?start=26&length=2)

Why only T0?

File: tests/gnmi/conftest.py (hunk 1 above)
Comment (ganglyu) [reply tests/gnmi/conftest.py:15]: https://github.com/sonic-net/sonic-mgmt/pull/7081
Zain's PR only supports x86 platform.
It's possible to test other platform, and we need to install GNMI client in sonic-mgmt container.

File: tests/gnmi/conftest.py (hunk 2 above)
Comment (ganglyu) [reply tests/gnmi/conftest.py:24]: I need GNOI client, but py_gnmicli.py only supports GNMI.

File: tests/gnmi/test_gnmi_appldb.py (hunk 4 above)
Comment (ganglyu) [reply tests/gnmi/test_gnmi_appldb.py:9]: https://github.com/sonic-net/sonic-mgmt/blob/master/tests/restapi/test_restapi.py#L16
Restapi only supports t0, GNMI will support t1?

Comment (qiluo-msft) [reply tests/gnmi/test_gnmi_appldb.py]: GNMI will support t1

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**